nexus/lib/nexus/room.ex

```diff
@@ -63,21 +66,16 @@ defmodule Nexus.Room do
   end
 
   @impl true
-  def handle_call({:mark_ready, id, pc, peer_track_specs}, _from, state)
+  def handle_call({:mark_ready, id, known_peer_ids}, _from, state)
       when is_map_key(state.pending_peers, id) do
     # FIXME: this seems like a crude way of alleviating certain RCs
     current_peer_ids = Map.keys(state.peers)
 
-    if current_peer_ids |> Enum.sort() != peer_track_specs |> Map.keys() |> Enum.sort() do
+    if Enum.sort(current_peer_ids) != Enum.sort(known_peer_ids) do
       {:reply, {:peer_mismatch, current_peer_ids}, state}
```

Comment:
Is this to ensure that there were no new peers when this specific peer was connecting to the server via WebRTC? If yes, we can solve this RC by something like this (this is code for the peer module):

```elixir
PubSub.subscribe(Nexus.PubSub, "room")
{:ok, other_peers} = Room.mark_ready(id)
```

Also, I don't see a lot of benefits of using PubSub here. We could solve this RC just by using Room to broadcast info about new peers, couldn't we?